### PR TITLE
Rohan i18nify org rstudio core

### PR DESCRIPTION
**Pull request by rohank07**
### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any Github issues that are related. 

This PR tackles hard-coded strings to i18n-ify in org.rstudio.core (excluding false-positives).
Issue: https://github.com/StatCan/daaas/issues/710

I did encounter hard-coded literals that were in Debug.log(). Would these strings be considered as user-facing to i18nify?  I have already i18nify'd them but just wanted to make sure.

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

This pull-request and future ones will tackle i18nify classes at package-level (e.g. org.rstudio.core) instead of the per-component approach (like we did for [all-options-menu](https://github.com/rstudio/rstudio/pull/9599))

Implementation follows the same pattern from #9485 as described in gwt/tools/i18n-helpers/README.md. Running the create_dev_locale.sh will create all the new "dev" locale from any new "en" properties files for easier visual confirmation. Refer to README for more information.
### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

Modify an entry in one of the new _en.properties files, and refresh RStudio with ant in devmode.
Run create_dev_locale.sh to clone all _en properties files, which will add a prefix of "@" to each constant and message. Then open RStudio using http://localhost:8787/?locale=dev and browse to the Options->* menus. You can also view in desktop mode by swapping the locale element value attribute from value="en" to value="dev" in RStudio.gwt.xml.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

cc @gtritchie

### CODE REVIEW

File: src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java [hunk 1]
```diff
@@ -784,7 +781,7 @@ public void refreshFocusableElements()
       ArrayList<Element> focusable = getFocusableElements();
       if (focusable.size() == 0)
       {
-         Debug.logWarning("No potentially focusable controls found in modal dialog");
+         Debug.logWarning(constants_.noFocusableControlsLog());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java:784]: Debug logging, don't localize.

File: src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java [hunk 2]
```diff
@@ -122,8 +124,8 @@ public boolean validate()
          fireEvent(new EnsureVisibleEvent());
          textBox_.getElement().focus();
          RStudioGinjector.INSTANCE.getGlobalDisplay().showErrorMessage(
-               "Error",
-               label_ + " must be a valid number.",
+               constants_.errorCaption(),
+               label_ + " " + constants_.rStudioGinjectorErrorMessage() ,
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java:128]: Use a placeholder in the string, not string concatenation.

File: src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java [hunk 3]
```diff
@@ -133,16 +135,16 @@ public boolean validate()
          if (minValue_ != null && intVal < minValue_)
          {
             RStudioGinjector.INSTANCE.getGlobalDisplay().showErrorMessage(
-                  "Error",
-                  label_ + " must be greater than or equal to " + minValue_ + ".",
+                    constants_.errorCaption(),
+                  label_ + " " + constants_.rStudioGinjectorGreaterThanError()  + minValue_ + ".",
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java:139]: Use placeholders in the string, not concatenation.

File: src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java [hunk 4]
```diff
@@ -133,16 +135,16 @@ public boolean validate()
...
                   textBox_);
             return false;
          }
          if (maxValue_ != null && intVal > maxValue_)
          {
             RStudioGinjector.INSTANCE.getGlobalDisplay().showErrorMessage(
-                  "Error",
-                  label_ + " must be less than or equal to " + maxValue_ + ".",
+                    constants_.errorCaption(),
+                  label_ + " " + constants_.rStudioGinjectorLessThanError() + maxValue_ + ".",
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/NumericValueWidget.java:147]: Use placeholders, not concatenation.

File: src/gwt/src/org/rstudio/core/client/widget/ProgressDialog.java [hunk 5]
```diff
@@ -176,7 +172,7 @@ protected void hideProgress()
       {
          operationStarted_ = false;
          announceCompletion(StringUtil.isNullOrEmpty(labelText_) ?
-            "Operation completed" : labelText_ + " completed");
+            constants_.operationCompletedText() : labelText_ + " " + constants_.completedText());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/ProgressDialog.java:175]: Use a placeholder, not concatenation.

File: src/gwt/src/org/rstudio/core/client/widget/ProgressSpinner.java [hunk 6]
```diff
@@ -60,7 +62,7 @@ public ProgressSpinner(int color)
       canvas_ = Canvas.createIfSupported();
       if (canvas_ == null)
       {
-         Debug.log("Can't create progress spinner (no HTML5 canvas support)");
+         Debug.log(constants_.progressSpinnerLog());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/ProgressSpinner.java:65]: Debug logging, don't localize.

File: src/gwt/src/org/rstudio/core/client/widget/TextBoxWithButton.java [hunk 7]
```diff
@@ -124,7 +126,7 @@ public TextBoxWithButton(String label,
       FlowPanel outer = new FlowPanel();
       if (label != null)
       {
-         assert existingLabel == null : "Invalid usage, cannot provide both label and existingLabel";
+         assert existingLabel == null : constants_.existingLabelMessage();
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/TextBoxWithButton.java:129]: Don't localize asserts.

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 8]
```diff
@@ -51,7 +52,7 @@ public void run()
    {
       if (running_)
       {
-         log("already running");
+         log(constants_.runningLabel());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:55]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 9]
```diff
@@ -68,21 +69,21 @@ public void run()
       {
          command.onExecute(() ->
          {
-            log("finished cmd");
+            log(constants_.finishedCmdLabel());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:72]: Logging, don't localize

Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:72]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 10]
```diff
@@ -68,21 +69,21 @@ public void run()
...
             countdown();
          });
       }
    }
 
    private void countdown()
    {
-      log("countdown");
+      log(constants_.countdownLabel());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:80]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 11]
```diff
@@ -68,21 +69,21 @@ public void run()
...
 
       commandsRemaining--;
 
       if (commandsRemaining <= 0)
       {
-         log("done");
+         log(constants_.doneLabel());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:86]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 12]
```diff
@@ -95,7 +96,7 @@ private void log(String label)
    {
       if (log_)
       {
-         Debug.log(hashCode() + " " + label + " size=" + commands_.size());
+         Debug.log(hashCode() + " " + label + " " + constants_.logSizeLabel() + commands_.size());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/ParallelCommandList.java:99]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/files/filedialog/FileDialog.java [hunk 13]
```diff
@@ -135,15 +137,15 @@ protected boolean shouldAccept()
...
+            showError(constants_.nonexistentFileMessage());
             return false;
          }
       }
       else
       {
          if (item.isDirectory())
          {
-            assert false : "This case should be covered by navigateIfDirectory";
+            assert false : constants_.navigateIfDirectoryMessage();
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/files/filedialog/FileDialog.java:148]: Don't localize asserts

File: src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java [hunk 14]
```diff
@@ -141,7 +143,7 @@ public void onResponseReceived(Request request,
                   {
                      String responseText = response.getText();
                      if (TRACE)
-                        Debug.log("Response: " + responseText);
+                        Debug.log(constants_.responseText() + responseText);
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java:146]: Logging, don't localize

File: src/gwt/src/org/rstudio/core/client/prefs/PreferencesDialogBase.java [hunk 15]
```diff
@@ -249,9 +251,9 @@ protected void restart(GlobalDisplay globalDisplay,
    {
       globalDisplay.showYesNoMessage(
             GlobalDisplay.MSG_QUESTION,
-            "Restart Required",
-            "You need to restart RStudio in order for these changes to take effect. " +
-                  "Do you want to do this now?",
+            constants_.restartRequiredCaption(),
+            constants_.restartRequiredMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/prefs/PreferencesDialogBase.java:255]: Use a single string here, not two concatenated together.

File: src/gwt/src/org/rstudio/core/client/prefs/PreferencesDialogBase.java [hunk 16]
```diff
@@ -267,9 +269,9 @@ protected void restartSession(GlobalDisplay display)
    {
       display.showYesNoMessage(
             GlobalDisplay.MSG_QUESTION,
-            "Restart Required",
-            "You need to restart the R session in order for these changes to take effect. " +
-            "Do you want to do this now?",
+            constants_.restartRequiredCaption(),
+              constants_.restartRequiredMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/prefs/PreferencesDialogBase.java:273]: Use a single string here, not two concatenated together.

File: src/gwt/src/org/rstudio/core/client/widget/LabeledTextBox.java [hunk 17]
```diff
@@ -85,7 +87,7 @@ public void disableAutoBehavior()
 
    public void setEnableSpellcheck(boolean enable)
    {
-      textBox_.getElement().setAttribute("spellcheck", enable ? "true" : "false");
+      textBox_.getElement().setAttribute(constants_.spellCheckAttribute(), enable ? "true" : "false");
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/LabeledTextBox.java:90]: Do not localize, this is an HTML attribute name.

**Review by gtritchie — CHANGES_REQUESTED** [review]

File: src/gwt/src/org/rstudio/core/client/ParallelCommandList.java [hunk 18]
```diff
@@ -40,7 +41,7 @@ public void addCommand(SerializedCommand command)
    {
       if (running_)
       {
-         log("cannot add - already running");
+         log(constants_.addCommandLabel());
```
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/core/client/ParallelCommandList.java]: No, mandate. We are targetting only end user facing UI. I was not sure if this text would be displayed to user.

File: src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java [hunk 19]
```diff
@@ -40,10 +40,7 @@
 import com.google.gwt.user.client.ui.Widget;
 
 import elemental2.dom.DomGlobal;
-import org.rstudio.core.client.Debug;
-import org.rstudio.core.client.ElementIds;
-import org.rstudio.core.client.Point;
-import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.*;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java:43]: Revert the wildcard import back to individual imports.

File: src/gwt/src/org/rstudio/core/client/widget/ProgressDialog.java [hunk 20]
```diff
@@ -33,11 +33,7 @@
 import com.google.gwt.user.client.Event.NativePreviewEvent;
 import com.google.gwt.user.client.ui.*;
 
-import org.rstudio.core.client.BrowseCap;
-import org.rstudio.core.client.ElementIds;
-import org.rstudio.core.client.HandlerRegistrations;
-import org.rstudio.core.client.Size;
-import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.*;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/core/client/widget/ProgressDialog.java:36]: Revert wildcard import back to individual imports.

**Review by gtritchie — CHANGES_REQUESTED** [review]
Getting close!

**Review by gtritchie — APPROVED** [review]
Looks good, thanks!

**Pull request merged**